### PR TITLE
SPR-117 Get Records Based on IDIR

### PR DESCRIPTION
**Pull request by dbarkowsky**
# Description

This PR includes the following proposed change(s):

- Add a toggle for admins on main page to just see their records
- Added a page that just shows an individual IDIR's records and an option on an individual record to get there.
- Corrected the Back and Update button redirects to go back to the correct page, depending on where they came from
- Added a navigate to the page's title, to take a user to the home page.
- Added checks to the requests API route for whether the user matches the record they're trying to access and if that user is an admin. If neither of these things are true, the request is denied with a 403. 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change
- [x] Dependencies added/removed

## Aspect(s) of Project Affected
- [x] Frontend
- [x] API
- [ ] Database
- [ ] Workflows
- [x] Documentation

## Best Practices Checklist
- [x] I have performed a self-review of my own code
- [x] My code follows the Airbnb React Style Guidelines
- [x] Documentation has been updated to reflect my changes
- [x] New and existing tests pass locally with my changes

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
Tested locally in dev and containers.

### CODE REVIEW

File: api/controllers/requests-api-controller.ts [hunk 1]
```diff
@@ -91,6 +91,11 @@ export const getAllRequests = async (req: Request, res: Response) => {
  */
 export const getRequestsByIDIR = async (req: Request, res: Response) => {
   const { minimal, idir } = req.query;
+
+  // Check if user's IDIR matches
+  const userInfo = getUserInfo(req.headers.authorization.split(' ')[1]); // Excludes the 'Bearer' part of token.
+  if (userInfo.idir_user_guid !== idir && !userInfo.client_roles?.includes('admin')) return res.status(403).send('Forbidden: User does not match requested IDIR.');
```
Comment (BradyMitch) [inline api/controllers/requests-api-controller.ts:97]: Shouldn't this be an OR not and AND? Meaning they either match the idir or are an admin?

Comment (dbarkowsky) [reply api/controllers/requests-api-controller.ts:97]: If it were an OR, then a user whose IDIR matches but isn't an admin wouldn't be able to see their own record. 
An admin who is looking at another user's record would also be denied. 

I need this denial to only come if the user's IDIR doesn't match AND they aren't an admin. It's a weird double negative, but I think that's what I need. 

...

I ended up doing a truth table after thinking about your comment. It looks like I could also do:
```js
if (!(userInfo.idir_user_guid === idir || userInfo.client_roles?.includes('admin')))
```
But then I have the NOT the whole thing, which I find harder to read.

File: api/controllers/requests-api-controller.ts [hunk 2]
```diff
@@ -177,6 +182,11 @@ export const updateRequestState = async (req: Request, res: Response) => {
   // Get previous state of request
   const existingRequest : WithId<RequestRecord> = await collection.findOne(filter);
 
+  if (existingRequest){
+    const userInfo = getUserInfo(req.headers.authorization.split(' ')[1]); // Excludes the 'Bearer' part of token.
+      if (userInfo.idir_user_guid !== existingRequest.idir && !userInfo.client_roles?.includes('admin')) return res.status(403).send('Forbidden: User does not match requested record.');
```
[the remark this answers is not in the document]
Comment (dbarkowsky) [reply api/controllers/requests-api-controller.ts:187]: Reply in previous comment.

File: api/controllers/requests-api-controller.ts (hunk 1 above)
Comment (BradyMitch) [reply api/controllers/requests-api-controller.ts:97]: There is definitely instances where using the (!( /* CONDITION */ )) makes sense and I think this is one of them.
You can also create

``` JavaScript
const idirMatchesUser = userInfo.idir_user_guid === idir;
const isAdmin = userInfo.client_roles?.includes('admin');

if (!(idirMatchesUser || isAdmin)) {}
```

Comment (BradyMitch) [reply api/controllers/requests-api-controller.ts:97]: Actually that may not work. You might have to do if idir doesnt match and not an admin like you had it

Comment (dbarkowsky) [reply api/controllers/requests-api-controller.ts]: I ended up breaking down the conditions into separate variables for clarity, and comments were added, but I still think the`!X && !Y` logic is clearer than `!(X || Y)` in this case, so I've left that intact.

Comment (BradyMitch) [reply api/controllers/requests-api-controller.ts]: @dbarkowsky wont that still not work for admins if the idir doesnt match?

**Review by BradyMitch — APPROVED** [review]
As long as you've tested that an admin can view other user's records I see this as complete.

**Pull request merged**